src/searchbar.cpp

```diff
@@ -114,6 +115,22 @@ void SearchBarLineEdit::hideSuggestions()
     m_completer.popup()->hide();
 }
 
+bool SearchBarLineEdit::eventFilter(QObject *, QEvent *event)
+{
+    if (!m_aboutToScrollPastEnd)
+        return false;
+
+    if (auto e = dynamic_cast<QKeyEvent *>(event))
```

Comment:
This time I checked [the documentation of `QObject::eventFilter()`](https://doc.qt.io/qt-5/qobject.html#eventFilter) as I was curious about the need for its first parameter). I saw in the example that a `dynamic_cast` is not used because there is a cheaper way - `QEvent::type()`.